batchspawner/batchspawner.py

```diff
@@ -471,7 +475,7 @@ class SlurmSpawner(UserEnvMixin,BatchSpawnerRegexStates):
     def parse_job_id(self, output):
         # make sure jobid is really a number
         try:
-            id = output.split(' ')[-1]
```

Comment:
The original version assumes that the SLURM job status output will always have the job ID come last. Yours assumes that the job ID will always be the fourth token. That seems like a more fragile assumption. Can you comment on the reason for this change?

Reply:
Sure, when you submit a job using the default cluster:

```bash
[bmooreii@login0b ~]$ sbatch hello.slurm 
Submitted batch job 914521
```

However, if you submit to a non-default cluster:

```bash
[bmooreii@login0b ~]$ sbatch --cluster=smp hello.slurm 
Submitted batch job 914522 on cluster smp
```

If you want to use the cluster flags (which I do). You need to use the 3rd entry not the -1st entry.

Reply:
Ah, okay then. Reading the manpage, it sounds like we should just be using the `--parsable` option for `sbatch` anyway to get guaranteed machine-readable output. Would you be willing to give that a try, since I don't have a SLURM available for testing? 

Also, can you provide the output of the `squeue` command as called in `SlurmSpawner.batch_query_command` to verify that we don't need to use the cluster name to query the job when running on a non-default cluster? Thanks!

Reply:
@mbmilligan 

I can definitely implement the `--parsable` flag. It will end up being more robust over time.

As for `SlurmSpawner.batch_query_command`, I still need to work on that. My current prod/dev jupyterhub setup isn't ideal for development.

Thanks for the comments!

Reply:
The `squeue` command must contain `-M {cluster}` when querying non-default clusters. I modify that in the configuration.

Reply:
@mbmilligan Additionally, `--parsable` is done. I will try to get JupyterHub 0.8.x working for additional testing.